### ISSUE TITLE
self.model has no attribute module in d2/train_net.py

### ISSUE THREAD

**Issue by uzck** [KaiyuYue/mgd#4]
I run your code and occur this error message, the `self.model` is an MGDistiller or SMDistiller object, and both of them don't have 'module' attribute, and I replace the `self.model.module` to `self.model`, the code works. It's right ?

**Comment by kaiyuyue** [comment]
Hmm, a little weird. As far as I know, I think the error is not expected. Once the whole model is wrapped by DDP [[MGD code in lines](https://github.com/KaiyuYue/mgd/blob/master/d2/train_net.py#L127-L136)] / [[official code in lines](https://github.com/facebookresearch/detectron2/blob/master/detectron2/engine/defaults.py#L286-L290)], it will have `module` attribute automatically. Which PyTorch version do you use?  I did training with PyTorch 1.4.0.

If the code can be run on GPUs after removing `module`, it largely works right because `self.model.module.{attribute_functions}` belong to the kind of sub-function to extract features, process flow matrix, etc., which works without gradients. It's okay without `module`. However, I still think something is wrong, and I cannot pinpoint what is wrong because I don't know your environment. You should check it deeply by yourself.

**Comment by uzck** [comment]
> Hmm, a little weird. As far as I know, I think the error is not expected. Once the whole model is wrapped by DDP [[MGD code in lines](https://github.com/KaiyuYue/mgd/blob/master/d2/train_net.py#L127-L136)] / [[official code in lines](https://github.com/facebookresearch/detectron2/blob/master/detectron2/engine/defaults.py#L286-L290)], it will have `module` attribute automatically. Which PyTorch version do you use? I did training with PyTorch 1.4.0.
> 
> If the code can be run on GPUs after removing `module`, it largely works right because `self.model.module.{attribute_functions}` belong to the kind of sub-function to extract features, process flow matrix, etc., which works without gradients. It's okay without `module`. However, I still think something is wrong, and I cannot pinpoint what is wrong because I don't know your environment. You should check it deeply by yourself.

I use a single RTX 2070 to train, so the problem maybe in here, and maybe you could share the commit id of detectron2 you used, it's still update now, and the current version is not compatible to your code.

**Comment by kaiyuyue** [comment]
Thanks.
